sklearn/compose/_column_transformer.py

```diff
@@ -597,6 +608,50 @@ def _get_transformer_list(estimators):
     return transformer_list
 
 
+def select_types(dtypes):
+    """Generate a column selector callable (type-based) to be passed to
+    `make_column_transformer` or `ColumnTransformer` in place of the column
+    selections. The returned callable will return a boolean mask for column
+    selection when applied to a dataframe.
+
+    Parameters
+    ----------
+    dtypes : list of column dtypes to be selected from the dataset.
+
+    Returns
+    -------
+    callable
+
+    Examples
+    --------
+    >>> import pandas as pd
+    >>> df = pd.DataFrame({
+    ...     'a': [1.5, 2.5],
+    ...     'b': ['dog', 'cat'],
+    ...     'c': ['big', 'small'],
+    ...     'd': [1, 2]
+    ... })
+    >>> df.dtypes
+    a    float64
+    b     object
+    c     object
+    d      int64
+    dtype: object
+    >>> type_selector = select_types([int, float])
+    >>> type_selector(df)
+    array([ True, False, False,  True])
+    """
+    def apply_dtype_mask(X, dtype):
+        if hasattr(X, 'dtypes'):
+            return X.dtypes == dtype
+
+    def get_dtype_masks(X):
+        masks = [apply_dtype_mask(X, t) for t in dtypes]
+        return masks
+
+    return lambda X: np.any(get_dtype_masks(X), axis=0)
```

Comment:
We cannot pickle closures. Please implement this as a class with `__call__` instead

Reply:
Yep, using a class with `__call__` will also make the code a lot cleaner.